impeller/renderer/backend/vulkan/android/ahb_texture_source_vk.cc

```diff
@@ -281,14 +282,19 @@ static TextureDescriptor ToTextureDescriptor(
   desc.mip_count = (ahb_desc.usage & AHARDWAREBUFFER_USAGE_GPU_MIPMAP_COMPLETE)
                        ? ahb_size.MipCount()
                        : 1u;
+  if (is_swapchain_image) {
+    desc.usage = TextureUsage::kRenderTarget;
+  }
+
   return desc;
 }
 
 AHBTextureSourceVK::AHBTextureSourceVK(
     const std::shared_ptr<Context>& p_context,
     struct AHardwareBuffer* ahb,
-    const AHardwareBuffer_Desc& ahb_desc)
-    : TextureSourceVK(ToTextureDescriptor(ahb_desc)) {
+    const AHardwareBuffer_Desc& ahb_desc,
+    bool is_swapchain_image)
```

Comment:
Consider inferring whether the image is a swapchain image based on the presence of AHARDWAREBUFFER_USAGE_COMPOSER_OVERLAY in the ahb_desc usage instead since that is already passed in. You can also perhaps add `impeller::android::HardwareBuffer::IsSwapchainImage` that does a similar usage check. That way, the API is nice and concise and the right™ thing is done based on usage.

Reply:
Done